### ISSUE TITLE
Mapping fails while reading an entity with GeoProperty

### ISSUE THREAD

**Issue by techbeansaus** [wistefan/ngsi-ld-java-mapping#50]
Hi Guys, 

I have an entity called Precinct as defined below.

```
package city.ui.models;

import java.net.URI;
import io.github.wistefan.mapping.annotations.*;

@MappingEnabled (entityType="Precinct")
public class Precinct {
    private static final String ENTITY_TYPE = "Precinct";
    private URI id;
    private String description;
    private String name;
    private GeoPolygon location;
    private String refTenant;

    public Precinct(String id) {
        this.id = URI.create(id); 
    }

    @EntityId
    @RelationshipObject
    @DatasetId
    public URI getId() {
        return id;
    }

    @EntityType
    public String getType() {
        return ENTITY_TYPE;
    }

    @AttributeGetter(value = AttributeType.PROPERTY, targetName = "description")
    public String getDescription() {
        return description;
    }

    @AttributeSetter(value = AttributeType.PROPERTY, targetName = "description")
    public void setDescription(String description) {
        this.description = description;
    }

    @AttributeGetter(value = AttributeType.PROPERTY, targetName = "name")
    public String getName() {
        return name;
    }

    @AttributeSetter(value = AttributeType.PROPERTY, targetName = "name")
    public void setName(String name) {
        this.name = name;
    }

    @AttributeGetter(value = AttributeType.GEO_PROPERTY, targetName = "location")
    public GeoPolygon getLocation() {
        return location;
    }

    @AttributeSetter(value = AttributeType.GEO_PROPERTY, targetName = "location", targetClass = GeoPolygon.class)
    public void setLocation(GeoPolygon location) {
        this.location = location;
    }

    @AttributeGetter(value = AttributeType.RELATIONSHIP, targetName = "refTenant")
    public String getRefTenant() {
        return refTenant;
    }

    @AttributeSetter(value = AttributeType.RELATIONSHIP, targetName = "refTenant", targetClass=Tenant.class)
    public void setRefTenant(String refTenant) {
        this.refTenant = refTenant;
    }
}
```

where GeoPolygon is defined as below

```
package city.ui.models;

import io.github.wistefan.mapping.annotations.MappingEnabled;
import lombok.Data;
@Data
public class GeoPolygon {
    private String type;
    private double[][][] coordinates;
}

```

I am able to create an ngsi ld entity from a json payload.

However, while I try to read the same entity and map it using the library, it fails. 

I can read from the context broker with no issues.

Entity which fails during mapping back to Java Object is as follows:

```
{
    "@context": [
        "https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld",
        "https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context-v1.6.jsonld"
    ],
    "id": "urn:ngsi-ld:Precinct:ba95b6d7-02a9-4bc5-97ed-ca7bb7444b53",
    "type": "Precinct",
    "name": {
        "type": "Property",
        "value": "Sample Location"
    },
    "description": {
        "type": "Property",
        "value": "This is a sample location description"
    },
    "location": {
        "type": "GeoProperty",
        "value": {
            "type": "Polygon",
            "coordinates": [
                [
                    [
                        0,
                        0
                    ],
                    [
                        4,
                        0
                    ],
                    [
                        4,
                        -2
                    ],
                    [
                        0,
                        -2
                    ],
                    [
                        0,
                        0
                    ]
                ]
            ]
        }
    }
}
```

Error is as follows:

`com.fasterxml.jackson.databind.exc.MismatchedInputException: Unexpected token (START_OBJECT), expected START_ARRAY: need Array value to contain `As.WRAPPER_ARRAY` type information for class org.fiware.ngsi.model.GeoPropertyVO
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 359] (through reference chain: org.fiware.ngsi.model.EntityVO["location"])`

Basically, Jackson fails to deserialize this entity to EntityVO object using the object mapper api.

```
EntityVO parsedEntity = OBJECT_MAPPER.readValue(jsonString, EntityVO.class);
```
Please let me know if I am missing anything.

**Comment by techbeansaus** [comment]
The issue is partly fixed by renaming "location" to anything but "location", "observationSpace" or "operationSpace". 
Also, there is another issue encountered once you rename the "location" attribute or the GeoPropertyVO attribute. The entityVOMapper.fromEntityVO() fails to map the EntityVO object to your Java POJO. The reason for that is as follows:

GeoPropertyVO is not an instance of PropertyVO and due to this reason, the handleProperty does not evaluate GeoPropertyVO.

`private <T> Mono<T> handleProperty(AdditionalPropertyVO propertyValue, T objectUnderConstruction, Method setter, Class<?> parameterType) {
        if(propertyValue instanceof GeoPropertyVO propertyVO){
            System.out.println("Geo Property");
            return invokeWithExceptionHandling(setter, objectUnderConstruction, objectMapper.convertValue(propertyVO.getValue(), parameterType));
        }
        if (propertyValue instanceof PropertyVO propertyVO)
            return invokeWithExceptionHandling(setter, objectUnderConstruction, objectMapper.convertValue(propertyVO.getValue(), parameterType));
        else {
            log.error("Mapping exception");
            return Mono.error(new MappingException(String.format("The attribute is not a valid property: %s ", propertyValue)));
        }
    }`

**Comment by wistefan** [comment]
Hi, if you found a fix, a PR would be very welcome:)

**Comment by techbeansaus** [comment]
> Hi, if you found a fix, a PR would be very welcome:)

I have raised a PR for this. Please review.
